### DISCUSSION TITLE
Divisionals and generals on remote control

### DISCUSSION THREAD

**Discussion by willeke1234** [GrandOrgue/grandorgue#1173] [Help]
A divisional or general is a trigger button if it's "empty" and a radio button if it's not. Is there a way for my remote control app to know if a division stays pressed when used?

**Comment by davidgritter** [comment]
As far as I can tell, using the grandorgue demo organ as an example, the generals on the main panel stay depressed independent of whether or not they have content.  If they have no content they don't affect the stops in any way.  The generals on the general panel behave the same way, staying pressed independent of their content, with the first few generals on the general panel duplicating those on the main panel. Generals on the main panel get reset by any of the generals on the generals panel. 

The Divisionals panel behaves differently.   Only a divisional that has been set with content will stay activated. Pressing any other divisional will have no effect on the stops and it will not show as depressed, even momentarily.  Once a divisional is programmed it will affect the stops and will stay depressed.  It will not however reset the depressed state of any of the generals.  Pressing another divisional, even if it is not programmed will release the programmed divisional that's set down.
I note that their appears to be a difference in the physical appearance of the programmed divisionals:  when unprogrammed their is no background behind the button; when programmed there appears to a small octagonal outline around the button.  This outline also seems to appear around all the  generals.

In any case, generals or divisionals,  it appears that right clicking and setting a 9x note message in the send tab will always send a note on message when the button is pressed (and stays depressed) and a note off message when it is cancelled, even if cancelled by an umprogrammed divisonal.  This midi message always seems to reliably indicate the state of the piston, even when performing a panic operation from the drop down menus.

**Comment by willeke1234** [comment]
The programmable generals on the Demo organ are setters and they stay pressed.
I also noticed that a divisional/general should get back to unpressed when the MIDI event is not sent back. Doing B when A doesn't happen isn't easy.
